crates/pallets/starknet/src/utils.rs

```diff
@@ -0,0 +1,83 @@
+use alloc::string::String;
+use alloc::vec::Vec;
+
+use blockifier::block_context::BlockContext;
+use blockifier::execution::entry_point::CallInfo;
+use blockifier::transaction::objects::{TransactionExecutionInfo, TransactionExecutionResult};
+use frame_support::storage;
+use mp_felt::Felt252Wrapper;
+use mp_simulations::{ExecuteInvocation, RevertedInvocation};
+use mp_transactions::execution::Execute;
+use mp_transactions::UserTransaction;
+use sp_runtime::DispatchError;
+
+use crate::blockifier_state_adapter::BlockifierStateAdapter;
+use crate::{pallet, Error};
+
+pub fn execute_txs_and_rollback<T: pallet::Config>(
+    txs: &Vec<UserTransaction>,
+    block_context: &BlockContext,
+    chain_id: Felt252Wrapper,
+    is_query: bool,
+    disable_validation: bool,
+    disable_fee_charge: bool,
+) -> Result<Vec<TransactionExecutionResult<TransactionExecutionInfo>>, Error<T>> {
+    let mut execution_results = vec![];
+    storage::transactional::with_transaction(|| {
+        for tx in txs {
+            let result = match tx {
+                UserTransaction::Declare(tx, contract_class) => {
+                    let executable =
+                        tx.try_into_executable::<T::SystemHash>(chain_id, contract_class.clone(), is_query);
+
+                    match executable {
+                        Err(err) => Err(err),
+                        Ok(executable) => executable.execute(
+                            &mut BlockifierStateAdapter::<T>::default(),
+                            block_context,
+                            is_query,
+                            disable_validation,
+                            disable_fee_charge,
+                        ),
+                    }
+                }
```

Comment:
This is a map, written as a match. I think you can use map, and chain it with the previous line